app-backend/db/src/main/scala/TaskDao.scala

```diff
@@ -0,0 +1,208 @@
+        fr"task_actions",
+        Nil
+      )
+      .filter(fr"task_id = $taskId")
+      .list
+  }
+
+  def getTaskWithActions(taskId: UUID): ConnectionIO[Option[Task.TaskFeature]] =
+    (for {
+      task <- OptionT(getTaskById(taskId))
+      actions <- OptionT.liftF(getTaskActions(task.id))
+    } yield { task.toGeoJSONFeature(actions) }).value
+
+  def unsafeGetTaskWithActions(taskId: UUID): ConnectionIO[Task.TaskFeature] =
+    for {
+      task <- unsafeGetTaskById(taskId)
+      actions <- getTaskActions(task.id)
+    } yield { task.toGeoJSONFeature(actions) }
+
+  def updateTask(
+      taskId: UUID,
+      updateTask: Task.TaskFeatureCreate,
+      user: User
+  ): ConnectionIO[Option[Task.TaskFeature]] =
+    (for {
+      initial <- OptionT(getTaskById(taskId))
+      _ <- OptionT.liftF(updateF(taskId, updateTask).update.run)
+      _ <- OptionT.liftF(
+        appendAction(
+          taskId,
+          initial.status,
+          updateTask.properties.status,
+          user.id
+        )
+      )
+      withActions <- OptionT(getTaskWithActions(taskId))
+    } yield withActions).value
+
+  def deleteTask(taskId: UUID): ConnectionIO[Int] =
+    query.filter(taskId).delete
+
+  def listTasks(
+      queryParams: TaskQueryParameters,
+      pageRequest: PageRequest
+  ): ConnectionIO[PaginatedGeoJsonResponse[Task.TaskFeature]] =
+    for {
+      paginatedResponse <- Dao
+        .QueryBuilder[Task](listF, joinTableF, Nil)
+        .filter(queryParams)
+        .page(pageRequest)
+      withActions <- paginatedResponse.results.toList traverse { feat =>
+        unsafeGetTaskWithActions(feat.id)
+      }
+    } yield {
+      PaginatedGeoJsonResponse(
+        paginatedResponse.count,
+        paginatedResponse.hasPrevious,
+        paginatedResponse.hasNext,
+        paginatedResponse.page,
+        paginatedResponse.pageSize,
+        withActions
+      )
+    }
+
+  def toFragment(
+      tfc: Task.TaskFeatureCreate,
+      user: User
+  ): Fragment = {
+    fr"""(
+        ${UUID.randomUUID}, ${Instant.now}, ${user.id}, ${Instant.now}, ${user.id},
+        ${tfc.properties.projectId}, ${tfc.properties.projectLayerId}, ${tfc.properties.status},
+        null, null, ${tfc.geometry}
+    )"""
+  }
+
+  def insertTasks(
+      taskFeatureCollection: Task.TaskFeatureCollectionCreate,
+      user: User
+  ): ConnectionIO[Task.TaskFeatureCollection] = {
+    val featureInserts = taskFeatureCollection.features map {
+      toFragment(_, user)
+    }
+    featureInserts.toNel map { inserts =>
+      (insertF ++ fr"VALUES " ++ inserts.intercalate(fr",")).update
+        .withGeneratedKeys[Task](
+          "id",
+          "created_at",
+          "created_by",
+          "modified_at",
+          "modified_by",
+          "project_id",
+          "project_layer_id",
+          "status",
+          "locked_by",
+          "locked_on",
+          "geometry"
+        )
+        .compile
+        .toList map { (xs: List[Task]) =>
```

Comment:
I am nitpicking here, but I think it would be better if we could use a variable name like `tasks` or something alike.